plutus-core/plutus-ir/test/transform/retainedSize/datatypeLiveConstr.golden

```diff
@@ -0,0 +1,13 @@
+(let
+  (nonrec)
+  (datatypebind
+    (datatype
+      $1$
+      (tyvardecl $4$ Maybe (fun (type) (type)))
+      (tyvardecl $2$ a (type))
+      match_Maybe
+      (vardecl $17$ Nothing [ Maybe a ]) (vardecl $6$ Just (fun a [ Maybe a ]))
+    )
+  )
+  Nothing
+)
```

Comment:
`Nothing` is referenced, so it retains all of the data type (see the note), i.e. the `17` weight.

Reply:
Can you put these comments above the listing for the test case in `TransformSpec.hs`. They're too good to lose!